### PR TITLE
feat: support different justification category in CLI

### PR DESCRIPTION
**Pull request by yolocs**
Fix: https://github.com/abcxyz/jvs/issues/278

But should we consider a command to list available category? That will require a new JVS API...

### CODE REVIEW

File: pkg/cli/token_create.go [hunk 1]
```diff
@@ -125,7 +134,22 @@ func (c *TokenCreateCommand) Flags() *cli.FlagSet {
...
+		Name:    "category",
+		Target:  &c.flagCategory,
+		EnvVar:  "JVSCTL_CATEGORY",
+		Default: jvspb.DefaultJustificationCategory,
+		Usage:   `The justification category.`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "justification",
+		Target:  &c.flagExplanation,
```
Comment (sqin2019) [inline pkg/cli/token_create.go:150]: flagJustificationText?

File: pkg/cli/token_create.go [hunk 2]
```diff
@@ -185,8 +209,16 @@ func (c *TokenCreateCommand) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("unexpected arguments: %q", args)
 	}
 
-	if c.flagExplanation == "" {
-		return fmt.Errorf("explanation is required")
+	if c.flagExplanation != "" {
+		fmt.Fprintln(c.Stderr(), `WARNING: "explanation" flag is DEPRECATED and will be removed in a future release. Use "justification" flag instead.`)
```
Comment (sethvargo) [inline pkg/cli/token_create.go:213]:
```suggestion
		c.Errf(`WARNING: the "-explanation" flag is deprecated and will be removed in a future release. Use "-justification" instead.`)
```

File: pkg/cli/token_create.go (hunk 1 above)
Comment (sethvargo) [inline pkg/cli/token_create.go:150]:
```suggestion
		Target:  &c.flagJustificationText,
```

**Review by sethvargo — CHANGES_REQUESTED** [review]

File: pkg/cli/token_create.go [hunk 3]
```diff
@@ -125,7 +134,22 @@ func (c *TokenCreateCommand) Flags() *cli.FlagSet {
 		Target:  &c.flagExplanation,
 		Aliases: []string{"e"},
 		Example: "Debugging ticket #123",
-		Usage:   `A reason for the action.`,
+		Usage:   `DEPRECATED. Use "justification" flag instead. This flag will be removed in a future release.`,
```
Comment (sethvargo) [inline pkg/cli/token_create.go:137]: I'd probably leave the message as-is and hide the flag: https://github.com/abcxyz/pkg/blob/05a9df35d989594e4f242e397f29d96f3fc0b041/cli/flags.go#L542

```suggestion
Hidden: true,
Usage: `A reason for the action.`,
```

**Review by sethvargo — APPROVED** [review]

**Pull request merged**